### PR TITLE
Fix: #1350 Icon box links not working properly

### PR DESCRIPTION
**Pull request by frimpongopoku**
####  Summary / Highlights

I checked and found that we are actually checking for all the weird cases for user input. Looks like the fault was from `__is_custom_site` -- it cant be trusted. So I just implemented a better check for the app to accurately determine when we are on a MassEnergize domain... then show the appropriate link. (i.e. prefix with subdomain) 

#### Requirements

* [x] I've read and understood the [Contributing Guidelines](/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've tested my changes manually.
* [ ] I've added unit tests to my PR.

##### Transparency ([Project board](https://github.com/orgs/massenergize/projects/7/views/2))
* [x] I've given my PR a meaningful title.
* [x] I linked this PR to the relevant issue.
* [ ] I moved the linked issue from the "Sprint backlog" to "In progress" when I started this.
* [ ] I moved the linked issue to "QA Verification" now that it is ready to merge.

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Performance improvement
- [ ] Feature removal
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch or to a feature branch, _not_ the `main` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the
  issue number)
- [ ] All tests are
  passing: 
- [ ] New/updated tests are included

**Other information: Any otherthing we need to know**

### CODE REVIEW

File: src/components/Pages/HomePage/HomePage.js [hunk 1]
```diff
@@ -79,8 +79,11 @@ class HomePage extends React.Component {
     const { __is_custom_site, community } = this.props;
     const { subdomain } = community || {};
 
-    const prefix = !__is_custom_site && subdomain ? `/${subdomain}` : "";
-    
+    const { hostname } = new URL(window.location.href);
+    const isNotCustom = URLS.NONE_CUSTOM_WEBSITE_LIST.has(hostname);
+
+    const prefix = isNotCustom && subdomain ? `/${subdomain}` : "";
```
Comment (Opoku-Agyemang) [inline src/components/Pages/HomePage/HomePage.js:85]: do you mean suffix?
```suggestion
    const suffix = isNotCustom && subdomain ? `/${subdomain}` : "";
```

Comment (frimpongopoku) [reply src/components/Pages/HomePage/HomePage.js:85]: No, I meant prefix. In concatenation, we add this part, before the actual link the user provides...  Why suffix?

File: src/components/Pages/HomePage/HomePage.js [hunk 2]
```diff
@@ -302,6 +305,7 @@ class HomePage extends React.Component {
                 title="Get started - See your local options!"
                 boxes={iconQuickLinks}
                 prefix={prefix}
+                is_custom_site={__is_custom_site}
```
Comment (BradHN1) [inline src/components/Pages/HomePage/HomePage.js:308]: not used - delete

**Review by BradHN1 — APPROVED** [review]
LGTM - wiith one clean-up

**Pull request merged**